core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -130,8 +130,7 @@ object Invoker {
         if (config.zookeeperHosts.startsWith(":") || config.zookeeperHosts.endsWith(":")) {
           abort(s"Must provide valid zookeeper host and port to use dynamicId assignment (${config.zookeeperHosts})")
         }
-        val invokerName = cmdLineArgs.name.getOrElse(config.invokerName)
-        if (invokerName.trim.isEmpty) {
+        if (invokerName.isEmpty || invokerName.get.trim.isEmpty) {
```

Comment:
this check is only done when the id is missing - so it is possible to have a name that is an empty (`""` or `" "`) string accepted as the invoker name (only `Option(null) == None`).

i don't think that's intended.

Reply:
ah, got it - I added another `isEmpty` check as
```
val invokerName = cmdLineArgs.name.orElse(if (config.invokerName.trim.isEmpty) None else Some(config.invokerName))
```
so, use the `--name` arg, OR config.invokerName (if its not empty)